src/components/templates/product-table/index.tsx

```diff
@@ -0,0 +1,543 @@
+    case "rejected":
+      return "danger"
+    case "draft":
+    default:
+      return "default"
+  }
+}
+
+type ProductTableProps = {}
+
+const ProductTable: React.FC<ProductTableProps> = () => {
+  const [offset, setOffset] = useState(0)
+  const [deleteProduct, setDeleteProduct] = useState(undefined)
+  const [collectionsList, setCollectionsList] = useState<string[]>([])
+  const [limit, setLimit] = useState(20)
+  const [query, setQuery] = useState("")
+  const [currentPage, setCurrentPage] = useState(0)
+  const [numPages, setNumPages] = useState(0)
+  const [showList, setShowList] = useState(true)
+  const toaster = useToaster()
+  const [tags, setTags] = useState(null)
+
+  const [statusFilter, setStatusFilter] = useState({
+    open: false,
+    filter: null,
+  })
+  const [collectionFilter, setCollectionFilter] = useState({
+    open: false,
+    filter: null,
+  })
+  const [tagsFilter, setTagsFilter] = useState({
+    open: false,
+    filter: null,
+    invalidTagsMessage: null,
+  })
+
+  const resetFilters = () => {
+    setStatusFilter({
+      open: false,
+      filter: null,
+    })
+    setCollectionFilter({
+      open: false,
+      filter: null,
+    })
+    setTagsFilter({
+      open: false,
+      filter: null,
+      invalidTagsMessage: null,
+    })
+  }
+
+  const clearFilters = () => {
+    resetFilters()
+    setOffset(0)
+    replaceQueryString({})
+  }
+
+  const submitFilters = async () => {
+    const collectionIds = collectionFilter.filter
+      ? collectionFilter.filter
+          .split(",")
+          .map((cf) => collections.find((c) => c.title === cf)?.id)
+          .filter(Boolean)
+          .join(",")
+      : null
+
+    const tagIds = tagsFilter.filter
+      ? tagsFilter.filter
+          .map((tag) => tag.trim())
+          .map((tag) => tags?.find((t) => t.value === tag)?.id)
+          .filter(Boolean)
+          .join(",")
+      : null
+
+    const urlObject = {
+      "status[]": statusFilter.open ? statusFilter.filter : null,
+      "collection_id[]": collectionFilter.open ? collectionIds : null,
+      "tags[]": tagsFilter.open && tagIds?.length > 0 ? tagIds : null,
+    }
+
+    const url = { ...removeNullish(urlObject) }
+
+    replaceQueryString(url)
+  }
+
+  const toggleFilterTags = async (tagsFilter) => {
+    if (!tags) {
+      const tagsResponse = await Medusa.products.listTagsByUsage()
+      setTags(tagsResponse.data.tags)
+    }
+
+    const invalidTags = tagsFilter.filter?.filter((tag) =>
+      tags.every((t) => t.value !== tag)
+    )
+
+    tagsFilter.invalidTagsMessage =
+      invalidTags?.length > 0
+        ? invalidTags?.length === 1
+          ? `${invalidTags[0]} is not a valid tag`
+          : `${invalidTags} are not valid tags`
+        : null
+
+    setTagsFilter(tagsFilter)
+  }
+
+  const filtersOnLoad = qs.parse(
+    window.location.search.charAt(0) === "?"
+      ? window.location.search.substring(1)
+      : window.location.search
+  )
```

Comment:
Filters on load work really well, but it would be great to add the filters being applied on load to the custom filter count. 

So e.g. if I send you the following URL with the collection filter (shown in image below), it will show as "Custom filter: **1**", when another person opens it.

![image](https://user-images.githubusercontent.com/59018053/151019013-82953879-82fd-4e72-9660-70592b093327.png)